include/psa/crypto.h

```diff
@@ -2904,7 +2904,7 @@ psa_status_t psa_generator_read(psa_crypto_generator_t *generator,
  *         If the key is persistent, the key material and the key's metadata
  *         have been saved to persistent storage.
  * \retval #PSA_ERROR_INSUFFICIENT_CAPACITY
- *                          There were fewer than \p output_length bytes
+ *                          There were fewer than \p bits * 8 bytes
```

Comment:
It looks like this should be `(bits +7) / 8` bytes.

Reply:
Pff, `bits * 8`, `bits / 8`…

It's actually `bits / 8` and not `(bits + 7) / 8` since `bits` is a multiple of 8 for symmetric keys.

But this function should also be specified for other key types. Implementations don't have to support every key type, but if they do then they should follow a standard. I've rewritten the specification of this function to define the behavior for every standard key type other than RSA.

Reply:
I pushed another commit to make the specification unambiguous for DH, DSA and ECC (Weierstrass). I chose a method that is codified by NIST, so that it's possible to be FIPS-compliant when implementing this function.